### PR TITLE
docs: Update SLSA verification commands

### LINKED ISSUE #14431: Improve SLSA verification command in the docs
The documentation https://argo-cd.readthedocs.io/en/stable/operator-manual/signed-release-assets/ shows how to verify the releases of argo-cd. The example commands using cosign https://argo-cd.readthedocs.io/en/stable/operator-manual/signed-release-assets/#verification-of-container-image-attestations don't properly verify the source repository of the build, and only verify the builder.

I would like to send a PR to improve the examples, as well as provide examples using slsa-verifier (already mentioned in the docs but does not have verification commands).

Thank you!

### PR DESCRIPTION
**Pull request by laurentsimon**
closes https://github.com/argoproj/argo-cd/issues/14431

Checklist:

* [X] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [X] The title of the PR states what changed and the related issues number (used for the release note).
* [X] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [X] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [X ] Does this PR require documentation updates?
* [X] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [X] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)). 
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [ ] I have added a brief description of why this PR is necessary and/or what this PR solves.

Please see [Contribution FAQs](https://argo-cd.readthedocs.io/en/latest/developer-guide/faq/) if you have questions about your pull-request.

### CODE REVIEW

File: docs/operator-manual/signed-release-assets.md [hunk 1]
```diff
@@ -60,40 +61,79 @@ The following checks were performed on each of these signatures:
...
---certificate-oidc-issuer https://token.actions.githubusercontent.com \
-quay.io/argoproj/argocd:v2.7.0 | jq
+# Get the immutable container image to prevent TOCTOU attacks.
+IMAGE=quay.io/argoproj/argocd:v2.7.0
+IMAGE="${IMAGE}@"$(crane digest "${IMAGE}")
+# Verify provenance, including the tag to prevent rollback attacks.
+slsa-verifier verify-image "$IMAGE" \
+    --source-uri github.com/argoproj/argo-cd \
+    --source-tag v2.7.0
+```
```
Comment (crenshaw-dev) [inline docs/operator-manual/signed-release-assets.md:80]: Can the docs briefly describe TOCTOU and rollback attacks, or link out to an external explanation?

File: docs/operator-manual/signed-release-assets.md [hunk 2]
```diff
@@ -60,40 +61,79 @@ The following checks were performed on each of these signatures:
...
+
+If you only want to verify up to the major or minor verion of the source repository tag (instead of the full tag), use the `--source-versioned-tag` which performs semantic versioning verification:
```
Comment (crenshaw-dev) [inline docs/operator-manual/signed-release-assets.md:82]: In what circumstance(s) might one want to only verify the major or minor version?

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]

**Comment by laurentsimon** [comment]
Let me know if there's something to do to fix pre-submits.

**Review by 34fathombelow — COMMENTED** [review]
@laurentsimon Thank you so much for the changes, they look great.  LGTM!  Just one request since this is a docs change.  I appreciate the update in the GO dependencies, but could we do this in a separate PR?  We will likely cherry-pick this back to our `release-2.7` branch.

**Comment by laurentsimon** [comment]
> > @laurentsimon Thank you so much for the changes, they look great. LGTM! Just one request since this is a docs change. I appreciate the update in the GO dependencies, but could we do this in a separate PR? We will likely cherry-pick this back to our `release-2.7` branch.
> 
> oops, totally. Was not supposed to be in the PR. I'll revert.

done.

**Review by 34fathombelow — APPROVED** [review]
LGTM, Thanks again!

File: docs/operator-manual/signed-release-assets.md (hunk 2 above)
Comment (crenshaw-dev) [reply docs/operator-manual/signed-release-assets.md:82]: Nah I think it's fine.  Thanks!

**Review by crenshaw-dev — APPROVED** [review]

**Pull request merged**